File: server/app/schemas/content_ops.py
from datetime import datetime
from typing import Any, Literal
from pydantic import BaseModel, Field


class ContentEntryTarget(BaseModel):
    topicType: Literal["daily", "evergreen", "review"] = "daily"
    entryId: str = Field(default="", max_length=64)
    h5Path: str = Field(default="", max_length=512)
    miniappPath: str = Field(default="", max_length=512)
    qrScene: str = Field(default="", max_length=64)
    officialAccountKeyword: str = Field(default="", max_length=32)


class ContentPublishPackageCreate(BaseModel):
    productKey: Literal["shenlun", "theory"]
    templateId: str
    sourceType: str = Field(min_length=1, max_length=32)
    sourceId: str = Field(min_length=1, max_length=32)
    sourceTitle: str = ""
    campaignKey: str = ""
    deepLink: str = ""
    entryTarget: ContentEntryTarget = Field(default_factory=ContentEntryTarget)
    slotValues: dict[str, str] = Field(default_factory=dict)
    variants: dict[str, dict[str, Any]] = Field(default_factory=dict)
    plannedAt: datetime | None = None


class ContentPublishStatusBody(BaseModel):
    status: Literal["draft", "teaching_review", "ops_review", "ready", "published", "rejected"]
    reviewNote: str = ""
    checklist: dict[str, bool] = Field(default_factory=dict)


class ContentPublishPackageUpdate(BaseModel):
    sourceTitle: str | None = None
    campaignKey: str | None = None
    deepLink: str | None = None
    entryTarget: ContentEntryTarget | None = None
    slotValues: dict[str, str] | None = None
    variants: dict[str, dict[str, Any]] | None = None
    plannedAt: datetime | None = None


class ContentPackageGenerateFromArticle(BaseModel):
    productKey: Literal["shenlun", "theory"]
    templateId: str
    articleId: str
    campaignKey: str = ""
    deepLink: str = ""
    entryTarget: ContentEntryTarget = Field(default_factory=ContentEntryTarget)
    plannedAt: datetime | None = None

File: server/admin-web/src/api/contentOps.ts
import http, { getData } from './http'

export type ContentOpsStatus = 'draft' | 'teaching_review' | 'ops_review' | 'ready' | 'published' | 'rejected'
export type ReviewStage = 'teaching' | 'operations'
export interface ReviewChecklistItem { key: string; label: string }
export interface ReviewStageConfig { key: ReviewStage; label: string; checklist: ReviewChecklistItem[] }
export interface ContentReviewRecord { id: string; stage: ReviewStage; decision: 'approved' | 'rejected'; checklist: Record<string, boolean>; note: string; reviewerId?: number; reviewerUsername: string; reviewerName: string; createdAt: string }
export interface ContentOpsTemplate { id: string; code: string; productKey: string; name: string; description: string; slots: string[]; channels: string[]; sortOrder: number; status: string }
export interface ContentVariant { title?: string; body?: string; [key: string]: unknown }
export interface ContentPackage { id: string; productKey: string; templateId: string; sourceType: string; sourceId: string; sourceTitle: string; campaignKey: string; deepLink: string; slotValues: Record<string, string>; variants: Record<string, ContentVariant>; reviewNote: string; reviewHistory: ContentReviewRecord[]; status: ContentOpsStatus; plannedAt?: string; publishedAt?: string; createdAt: string; updatedAt: string }
export interface ContentPackageExport { schemaVersion: string; generatedAt: string; template: ContentOpsTemplate; package: ContentPackage; channels: Array<{ channel: string; content: ContentVariant; deepLink: string; plannedAt?: string; manualPublishRequired: boolean }>; checklist: string[] }
export interface ContentOpsOverview { windowDays: number; scheduledCount: number; readyInventory: number; reviewBacklog: number; unplannedDrafts: number; productMix: { shenlun: number; theory: number }; statusCounts: Record<string, number>; alerts: Array<{ level: 'info' | 'warning'; code: string; message: string }>; healthy: boolean }
export interface ContentReferencePlatform { key: string; name: string; sourceStatus: string; sourceStatusLabel: string; format: string; structure: string[]; deliverables: string[]; example: { title: string; hook: string; sections: string[] } }
export interface ContentReferenceLibrary { schemaVersion: string; updatedAt: string; title: string; sources: Array<{ kind: string; label: string; path: string }>; platforms: ContentReferencePlatform[] }

export const fetchContentTemplates = (productKey?: string) => getData<ContentOpsTemplate[]>(http.get('/admin/content-ops/templates', { params: { productKey } }))
export const fetchContentReviewConfig = () => getData<{ stages: ReviewStageConfig[] }>(http.get('/admin/content-ops/review-config'))
export const fetchContentReferenceLibrary = () => getData<ContentReferenceLibrary>(http.get('/admin/content-ops/reference-library'))
export const fetchContentPackages = (params?: { productKey?: string; status?: string }) => getData<ContentPackage[]>(http.get('/admin/content-ops/packages', { params }))
export const fetchContentOpsOverview = (days = 7) => getData<ContentOpsOverview>(http.get('/admin/content-ops/overview', { params: { days } }))
export const createContentPackage = (data: Record<string, unknown>) => getData<ContentPackage>(http.post('/admin/content-ops/packages', data))
export const generateContentPackageFromArticle = (data: Record<string, unknown>) => getData<ContentPackage>(http.post('/admin/content-ops/packages/generate-from-article', data))
export const updateContentPackage = (id: string, data: Record<string, unknown>) => getData<ContentPackage>(http.put(`/admin/content-ops/packages/${id}`, data))
export const updateContentPackageStatus = (id: string, status: ContentOpsStatus, reviewNote = '', checklist: Record<string, boolean> = {}) => getData<ContentPackage>(http.post(`/admin/content-ops/packages/${id}/status`, { status, reviewNote, checklist }))
export const exportContentPackage = (id: string) => getData<ContentPackageExport>(http.get(`/admin/content-ops/packages/${id}/export`))
